knowage-vue/src/modules/documentBrowser/DocumentBrowser.vue

```diff
@@ -0,0 +1,147 @@
+                return
+            }
+
+            const id = this.tabs[this.activeIndex - 1].item ? this.tabs[this.activeIndex - 1].item.id : 'new-dashboard'
+            this.$router.push('/document-browser/document-execution/' + id)
+        },
+        onItemSelect(payload: any) {
+            this.tabs.push(payload)
+            const id = payload.item ? payload.item.id : 'new-dashboard'
+            this.$router.push('/document-browser/document-execution/' + id)
+            this.activeIndex = this.tabs.length
+        },
+        toggle(event: any) {
+            this.createMenuItems()
+            const menu = this.$refs.menu as any
+            menu.toggle(event)
+        },
+        createMenuItems() {
+            this.menuItems = []
+            this.menuItems.push({ label: this.$t('documentBrowser.closeCurrentDocument'), command: () => this.closeDocumentConfirm('current') })
+            this.menuItems.push({ label: this.$t('documentBrowser.closeOtherDocuments'), command: () => this.closeDocumentConfirm('other') })
+            this.menuItems.push({ label: this.$t('documentBrowser.closeDocumentRight'), command: () => this.closeDocumentConfirm('right') })
+            this.menuItems.push({ label: this.$t('documentBrowser.closeAllDocuments'), command: () => this.closeDocumentConfirm('all') })
+        },
+        closeDocumentConfirm(mode: string) {
+            this.$confirm.require({
+                message: this.$t('documentBrowser.closeDocumentConfirmMessage'),
+                header: this.$t('documentBrowser.closeDocumentConfirmTitle', { numberOfDocuments: this.getNumberOfDocuments(mode) }),
+                accept: () => this.closeDocument(mode)
+            })
+        },
+        getNumberOfDocuments(mode: string) {
+            switch (mode) {
+                case 'current':
+                    return 1
+                case 'other':
+                    return this.tabs.length - 1
+                case 'right':
+                    return this.tabs.length - this.activeIndex
+                case 'all':
+                    return this.tabs.length
+            }
+        },
+        closeDocument(mode: string) {
+            switch (mode) {
+                case 'current':
+                    this.tabs.splice(this.activeIndex - 1, 1)
+                    this.activeIndex = 0
+                    this.$router.push('/document-browser')
+                    break
+                case 'other':
+                    this.tabs = [this.tabs[this.activeIndex - 1]]
+                    this.activeIndex = 1
+                    break
+                case 'right':
+                    this.tabs.splice(this.activeIndex)
+                    break
+                case 'all':
+                    this.$router.push('/document-browser')
+                    this.tabs = []
+                    this.activeIndex = 0
+            }
+        }
+    }
+})
+</script>
+
+<style lang="scss">
```

Comment:
Remember to use scoped or use less generic class-names (.tab-container can wbe shared somewhere else.). If those classes are needed elsewhere just put them in common css